x/poolmanager/router.go

```diff
@@ -548,6 +548,43 @@ func (k Keeper) AllPools(
 	return sortedPools, nil
 }
 
+// ListPoolsByDenom returns all pools by denom sorted by their ids
+// from every pool module registered in the
+// pool manager keeper.
+func (k Keeper) ListPoolsByDenom(
+	ctx sdk.Context,
+	Denom string,
+) ([]types.PoolI, error) {
+	less := func(i, j types.PoolI) bool {
+		return i.GetId() < j.GetId()
+	}
+	//	Allocate the slice with the exact capacity to avoid reallocations.
+	poolCount := k.GetNextPoolId(ctx)
+	sortedPools := make([]types.PoolI, 0, poolCount)
```

Comment:
nit: are we reallocating here? since next `k.GetNextPoolId` would give an id that is one bigger than the number of existing pools, not ones with specific denom?